File: internships/yandex/mobile/sirius/3.py
s = input()

arr = [s[0]]
p = 1

while p <= len(s):

    c = 0
    while p + c <= len(s) and s[p:p+c] in arr:
        c += 1

    arr.append(s[p:p+c])
    p += c

arr.pop(1)
print(*arr)

File: internships/yandex/mobile/sirius/dart/3.dart
import 'dart:io';

void main() {
  String? s = stdin.readLineSync();
  List<String> arr = <String>[s![0]];

  if (s.length == 1) {
    print(s);
  } else {
    int p = 1;

    while (p < s.length) {
      int c = 0;

      while (p + c < s.length && arr.contains(s.substring(p, p + c))) {
        c += 1;
      }

      arr.add(s.substring(p, p + c));
      p += c;
    }

    arr.removeAt(1);
    print(arr.join(' '));
  }
}
